maven/lib/dependabot/maven/file_updater.rb

```diff
@@ -119,18 +120,74 @@ def update_pomfiles_for_property_change(pomfiles, req)
       end
       def update_version_in_file(dependency, file, previous_req, requirement)
         updated_content = T.must(file.content)
-
-        original_file_declarations(dependency, previous_req).each do |old_dec|
-          updated_content = updated_content.gsub(old_dec) do
-            updated_file_declaration(old_dec, previous_req, requirement)
+        original_file_declarations = original_file_declarations(dependency, previous_req)
+
+        if original_file_declarations.any?
+          # If the file already has a declaration for this dependency, we
+          # update the existing declaration with the new version.
+          original_file_declarations.each do |old_dec|
+            updated_content = updated_content.gsub(old_dec) do
+              updated_file_declaration(old_dec, previous_req, requirement)
+            end
           end
+        else
+          # If the file does not have a declaration for this dependency, we
+          # add a new declaration for it.
+          updated_content = add_new_declaration(updated_content, dependency, requirement)
         end
 
         raise "Expected content to change!" if updated_content == file.content
 
         updated_file(file: file, content: updated_content)
       end
 
+      sig do
+        params(
+          content: String,
+          dependency: Dependabot::Dependency,
+          requirement: T::Hash[Symbol, T.untyped]
+        ).returns(String)
+      end
+      def add_new_declaration(content, dependency, requirement) # rubocop:disable Metrics/AbcSize
```

Comment:
IMO the Metrics/AbcSize stuff is set way to low, it goes of for anything non-trivial.